addons/sale/models/sale_order_line.py

```diff
@@ -313,29 +313,32 @@ def _compute_no_variant_attribute_values(self):
     @api.depends('product_id')
     def _compute_name(self):
         for line in self:
-            if not line.product_id:
+            if not line.product_id and not (line.is_downpayment or line.display_type):
+                # don't compute line name for non down payment line with no product
                 continue
             lang = line.order_id._get_lang()
             if lang != self.env.lang:
                 line = line.with_context(lang=lang)
-            name = line._get_sale_order_line_multiline_description_sale()
-            if line.is_downpayment and not line.display_type:
-                context = {'lang': lang}
-                dp_state = line._get_downpayment_state()
-                if dp_state == 'draft':
-                    name = _("%(line_description)s (Draft)", line_description=name)
-                elif dp_state == 'cancel':
-                    name = _("%(line_description)s (Cancelled)", line_description=name)
-                else:
-                    invoice = line._get_invoice_lines().move_id
-                    if len(invoice) == 1 and invoice.payment_reference and invoice.invoice_date:
-                        name = _(
-                            "%(line_description)s (ref: %(reference)s on %(date)s)",
-                            line_description=name,
-                            reference=invoice.payment_reference,
-                            date=format_date(line.env, invoice.invoice_date),
-                        )
-                del context
+            if line.product_id:
+                line.name = line._get_sale_order_line_multiline_description_sale()
+                continue
+            context = {'lang': lang}
+            name = _("Down Payment")
```

Comment:
Hello @Feyensv :wave: 

Since removal of product of from down payment lines in [PR](https://github.com/odoo/odoo/pull/145887) down payment lines not contains product and because of that `_compute_name`  don't re-compute SOL name of down payment lines
I updated the code to recompute SOL name for `is_downpayment` but problem is I have to set name `Down Payent` statically before this version we were getting it from Product name
Is it okay? :thinking:

Also I tried keeping user updated SOL name https://github.com/odoo/odoo/pull/158206/commits/a99b24c06878dc61a19b1bbb51b555edef928152
Like if I added `whatever` posting invoice will update whole name and make it `Down Payment (ref...` And here it will update name to `whatever (Posted)`
But problem is it is bit ugly also will not work in other language and seems useless :cry: 
Can we try improving it in master? :thought_balloon: 

Thank you :innocent: